x-pack/plugins/ml/public/shared_components/single_metric_viewer/single_metric_viewer.tsx

```diff
@@ -0,0 +1,252 @@
+import { APP_STATE_ACTION } from '../../application/timeseriesexplorer/timeseriesexplorer_constants';
+import type { SingleMetricViewerServices, MlEntity } from '../../embeddables/types';
+import './_index.scss';
+
+const containerPadding = 10;
+const minElemAndChartDiff = 20;
+const RESIZE_THROTTLE_TIME_MS = 500;
+interface AppStateZoom {
+  from?: string;
+  to?: string;
+}
+
+const errorMessage = i18n.translate('xpack.ml.singleMetricViewerEmbeddable.errorMessage"', {
+  defaultMessage: 'Unable to load the ML single metric viewer data',
+});
+
+export type SingleMetricViewerSharedComponent = FC<SingleMetricViewerProps>;
+
+/**
+ * Only used to initialize internally
+ */
+export type SingleMetricViewerPropsWithDeps = SingleMetricViewerProps & {
+  coreStart: CoreStart;
+  pluginStart: MlDependencies;
+  mlServices: SingleMetricViewerServices;
+};
+
+export interface SingleMetricViewerProps {
+  bounds?: TimeRangeBounds;
+  selectedEntities?: MlEntity;
+  selectedDetectorIndex?: number;
+  functionDescription?: string;
+  selectedJobId: string | undefined;
+  /**
+   * Last reload request time, can be used for manual reload
+   */
+  lastRefresh?: number;
+  onLoading?: (isLoading: boolean) => void; // TODO: callback
+  onRenderComplete?: () => void;
+  onError?: (error: Error) => void;
+  uuid: string;
+}
+
+const SingleMetricViewerWrapper: FC<SingleMetricViewerPropsWithDeps> = ({
+  // Component dependencies
+  coreStart,
+  pluginStart,
+  mlServices,
+  // Component props
+  bounds,
+  functionDescription,
+  lastRefresh,
+  onError,
+  selectedDetectorIndex,
+  selectedEntities,
+  selectedJobId,
+  uuid,
+}) => {
+  const [chartWidth, setChartWidth] = useState<number>(0);
+  const [zoom, setZoom] = useState<AppStateZoom | undefined>();
+  const [selectedForecastId, setSelectedForecastId] = useState<string | undefined>();
+  const [selectedJob, setSelectedJob] = useState<MlJob | undefined>();
+  const [jobsLoaded, setJobsLoaded] = useState(false);
+
+  const isMounted = useRef(true);
+
+  const { mlApiServices, mlJobService, mlTimeSeriesExplorerService, toastNotificationService } =
+    mlServices;
+  const startServices = pick(coreStart, 'analytics', 'i18n', 'theme');
+  const datePickerDeps: DatePickerDependencies = {
+    ...pick(coreStart, ['http', 'notifications', 'theme', 'uiSettings', 'i18n']),
+    data: pluginStart.data,
+    uiSettingsKeys: UI_SETTINGS,
+    showFrozenDataTierChoice: false,
+  };
+
+  const previousRefresh = usePrevious(lastRefresh ?? 0);
+
+  useEffect(
+    function setUpJobsLoaded() {
+      async function loadJobs() {
+        try {
+          await mlJobService.loadJobsWrapper();
+          setJobsLoaded(true);
+        } catch (e) {
+          if (onError) {
+            onError(new Error(errorMessage));
+          }
+        }
+      }
+      if (isMounted.current === false) {
+        return;
+      }
+      loadJobs();
+
+      return () => {
+        isMounted.current = false;
+      };
```

Comment:
```suggestion
```

Reply:
updated in 6c61014b8a7ef4a3167a642e5f72f1938ea25c50